### PR TITLE
Bug 1284432 - Some changes to make cycling data succeed

### PR DESCRIPTION
**Pull request by wlach**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/2222)

### CODE REVIEW

File: treeherder/model/models.py [hunk 1]
```diff
@@ -706,6 +704,8 @@ class Meta:
             # this is intended to speed up queries for specific platform /
             # option collections on a push
             ('machine_platform', 'option_collection_hash', 'push'),
+            # speed up cycle data
+            ('repository', 'submit_time'),
```
Comment (edmorley) [inline treeherder/model/models.py]: I think this should be a standard index on only `submit_time`?
Whilst the current query breaks things down by repository, I don't think we should do that longer term. (I'm not saying that we should change that behaviour now, but a single index would fill both use-cases.)

File: treeherder/config/settings.py [hunk 2]
```diff
@@ -27,6 +27,9 @@ def server_supports_tls(url):
 
 # Default to retaining data for ~4 months.
 DATA_CYCLE_DAYS = env.int("DATA_CYCLE_DAYS", default=120)
+# Determines the number of jobs we try to delete per iteration when
+# cycling data
+DATA_CYCLE_CHUNK_SIZE = env.int("DATA_CYCLE_CHUNK_SIZE", default=1000)
```
Comment (edmorley) [inline treeherder/config/settings.py]: I think we may want to use a higher value than 1000 (from my testing running the command manually before), but we can always iterate.

File: treeherder/model/models.py (hunk 1 above)
Comment (wlach) [inline treeherder/model/models.py]: I think it actually makes more sense to cycle per-repository, as that lets us set a different expiry on them (i.e. in the future we might want to keep m-c data around longer than try).

File: treeherder/config/settings.py (hunk 2 above)
Comment (wlach) [inline treeherder/config/settings.py]: So in my testing we're still OOM'ing with this value: I guess going across all those foreign key relations can be expensive. Reducing it to 100 seems to work (it's steadily expiring data on inbound, at least).

Comment (wlach) [inline treeherder/config/settings.py]: More testing: it looks like an interval of 250 with no sleep time, has good behaviour on stage. Decent deletion rate (would have no problem keeping up with new jobs per day) while performance stays reasonable.

Comment (wlach) [inline treeherder/config/settings.py]: Later update: 250 still causes OOMs, though it takes a while. But 100 seems to succeed, and stage stays responsive. Should probably stick with that.

**Pull request merged**